libraries/AP_EFI/AP_EFI.cpp

```diff
@@ -257,6 +264,19 @@ void AP_EFI::get_state(EFI_State &_state)
     _state = state;
 }
 
+#if AP_EFI_SCRIPTING_ENABLED && AP_SCRIPTING_ENABLED
+
+void AP_EFI::handle_scripting(const EFI_State &efi_state)
```

Comment:
Rather than adding these "pass throughout" helpers you could add the whole backed as a `apobject`. That allows you to bypass the front end completely. You would just have a getter for that object. 

Alternatively you could add a singleton to the backed, however that would limit us to one instance. 

I think either method should save a little flash.

Reply:
I did that at first similar to Rishabh's rangefinder PR.  

Tridge preferred the method above and  this did seem a bit simpler.

Reply:
I don't think this is simpler, it removes the need for this upper level function at all. This approach pollutes the whole class with scripting stuff. The apobject keeps it in the backed.

I think apobject also works better for multiple backends, you just get a object for each and they work the same. This aproach we would have to add a instance number to this function. Of course you may not need multiple instances here, but we will for other stuff and it would be nice to have all the scripting backends working the same.

Reply:
i don't really see the advantage of the ap_object approach. I suggested this approach as it matches things like our handle_msg in GPS, follow, proximity, opendroneid etc. It seems like a nice simple approach for scripting to use
The thing that really chews the flash is the structure field bindings

Reply:
I would agree if multiple backend's needed the information, as they do in those examples. In this case only the scripting backed will ever implement this message. The goal is to get data from a script into the scripting backend, there is no need to go via the front end in that case. A example of this approach is the `get_rgb`  bindings for `AP_Notify/ScriptingLED`. In that case we go direct to the backend. Because there is only one we use a singleton whereas here we would need to do ap_object. We could make the same argument for the Quadplane bindings, again direct to the backend by passing AP_Vehcile and singleton rather than an ap_object. 

From the script users point of view it makes little difference, the key advantage to the ap_object method is that it keeps all the scripting stuff in the scripting backed. As the class level you only need a getter for the backend. Here each new method we might want to add in the future needs a function in both the class and the backed.